compiler/modules/front/codeGen.bal

```diff
@@ -1230,8 +1230,7 @@ function codeGenConditionalExpr(CodeGenContext cx, bir:BasicBlock block, Environ
 
 function codeGenExprForBoolean(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Expr expr) returns CodeGenError|BooleanExprEffect {
     var { result, block, narrowing } = check codeGenExpr(cx, bb, env, expr);
-    if result is bir:BooleanOperand {
-        // rest of the type checking is in the verifier
+    if result is bir:BooleanOperand && (result is boolean || t:isSubtypeSimple(result.semType, t:BOOLEAN)) {
```

Comment:
should be able to remove `result is bir:BooleanOperand &&` since it's redundant.